frontend/taipy-gui/src/utils/index.ts

```diff
@@ -105,14 +105,20 @@ export const getDateTimeString = (
     tz?: string,
     withTime: boolean = true
 ): string => {
-    if (withTime) {
-        return formatInTimeZone(
-            getDateTime(value) || "",
-            formatConf.forceTZ || !tz ? formatConf.timeZone : tz,
-            datetimeformat || formatConf.dateTime
-        );
+    const dateVal = getDateTime(value);
+    try {
+        if (withTime) {
+            return formatInTimeZone(
+                dateVal || "",
+                formatConf.forceTZ || !tz ? formatConf.timeZone : tz,
+                datetimeformat || formatConf.dateTime
+            );
+        }
+        return format(dateVal || 0, datetimeformat || formatConf.date);
```

Comment:
An easy thing that you could do first is simply to check if the format string (datetimeformat or formatConf.dateTime) contains a "D" or not. No need for fancy regexp here.
Add the useAdditionalDayOfYearTokens settings if yes, and we're all good.
What do you think?

Reply:
Then we may as well as the option all the time?